src/Entities/Player.cpp

```diff
@@ -1800,6 +1800,11 @@ bool cPlayer::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 	// Send the respawn packet:
 	if (a_ShouldSendRespawn && (m_ClientHandle != nullptr))
 	{
+		// Respawning in current dimension is illegal if wasn't dead, let's tour a bit...
```

Comment:
I would add a more descriptive comment so that people in the future would understand this. Something like:
```
When teleporting to a world of the same dimension as the current world, we must first send the respawn packet of some other dimension, otherwise the client will have visual issues. The real respawn packet is sent immediately afterwards below.
```